text/0111-auto-resource-detection.md

```diff
@@ -0,0 +1,178 @@
+# Automatic Resource Detection
+
+Introduce a mechanism to support auto-detection of resources.
+
+## Motivation
+
+Resource information, i.e. attributes associated with the entity producing
+telemetry, can currently be supplied to tracer and meter providers or appended
+in custom exporters. In addition to this, it would be useful to have a mechanism
+to automatically detect resource information from the host (e.g. from an
+environment variable or from aws, gcp, etc metadata) and apply this to all kinds
+of telemetry. This will in many cases prevent users from having to manually
+configure resource information.
+
+Note there are some existing implementations of this already in the SDKs (see
+[below](#prior-art-and-alternatives)), but nothing currently in the
+specification.
+
+## Explanation
+
+In order to apply auto-detected resource information to all kinds of telemetry,
+a user will need to configure which resource detector(s) they would like to run
+(e.g. AWS EC2 detector).
+
+If multiple detectors are configured, and more than one of these successfully
+detects a resource, the resources will be merged according to the Merge
+interface already defined in the specification, i.e. the earliest matched
+resource's attributes will take precedence. Each detector may be run in
+parallel, but to ensure deterministic results, the resources must be merged in
+the order the detectors were added.
+
+A default implementation of a detector that reads resource data from the
+`OTEL_RESOURCE` environment variable will be included in the SDK. The
+environment variable will contain of a list of key value pairs, and these are
+expected to be represented in a format similar to the [W3C
+Correlation-Context](https://github.com/w3c/correlation-context/blob/master/correlation_context/HTTP_HEADER_FORMAT.md#header-value),
+except that additional semi-colon delimited metadata is not supported, i.e.:
+`key1=value1,key2=value2`. If the user does not specify any resource, this
+detector will be run by default.
+
+Custom resource detectors related to specific environments (e.g. specific cloud
+vendors) must be implemented as packages separate to the core SDK, and users
+will need to import these separately.
+
+## Internal details
+
+As described above, the following will be added to the Resource SDK
```

Comment:
As stated in https://github.com/open-telemetry/oteps/pull/111#issuecomment-656034077, I'm not sure if this should be added directly to the Resource SDK spec. I thinklanguages should be encouraged to implement this as a separate, optional package.